containers/layout/Layout.tsx

```diff
@@ -2,39 +2,82 @@ import React from 'react';
 import Container from '@material-ui/core/Container';
 import { I18n, TFunction } from 'next-i18next';
 import Head from 'next/head';
-import { getTitleString } from 'lib/utils/strings';
 import Footer from 'components/footer/Footer';
 import useStyles from './LayoutStyles';
 import { Box, Typography } from '@material-ui/core';
 import OptionToggler from 'components/option-toggler/OptionToggler';
 import { Button } from 'components';
 import { useAuth } from 'hooks/useAuth';
 import { useRouter } from 'next/router';
-import { FirebaseError } from 'lib/types/auth';
 import Error from 'components/error';
+import { withTranslation } from 'i18n.config';
+import { layoutError } from 'apollo/cache';
+import { useReactiveVar } from '@apollo/client';
+import { getTitleString, getTitleStringFromPathname } from 'lib/utils/strings';
+import { gql, useQuery } from '@apollo/client';
+import { getShortName } from 'lib/utils/strings';
+
+const GET_TALENT_NAME = gql`
+  query GetTalentById($id: String!) {
+    getTalentById(id: $id) {
+      id
+      basicInfo {
+        id
+        name {
+          firstName
+          middleName
+          lastName
+        }
+      }
+    }
+  }
+`;
 
 export interface LayoutProps {
-  home?: boolean;
-  heading?: string;
-  title?: string | string[];
-  error: FirebaseError | null;
   children?: React.ReactNode;
   i18n?: I18n;
   t: TFunction;
 }
 
-const Layout = ({
-  home = false,
-  error = null,
-  children,
-  heading,
-  title,
-  i18n,
-  t,
-}: LayoutProps): React.ReactElement => {
+const Layout = ({ children, t, i18n }: LayoutProps): React.ReactElement => {
   const classes = useStyles();
   const auth = useAuth();
+  const reactiveError = useReactiveVar(layoutError);
   const router = useRouter();
+  const { pathname, query } = router;
+
+  const isHome = pathname === '/';
+  const loadBasicInfo = pathname.includes('/talents/[id]'); // dependent on how profile page is named
```

Comment:
maybe change variable name to isOnTalentPage
(and maybe just check wether path includes '/talents/'

Reply:
This would conflict with the `settings` page. As we have it now, we wouldn't want the same `title` to be applied for the `settings` page.
I think the ideal solution here would be: 
- check if we are on one the `/talents/...` pages
- if yes, we query the `basicInfo`
- we check the current `pathname` to see if we are on `profile` or `settings` or somewhere else
- then we use the `pathname` and the `basicInfo` to give a custom `title` for the specific `/talents/...` page

What do you think?